core/src/main/java/org/apache/carbondata/core/util/CleanFilesUtil.java

```diff
@@ -0,0 +1,179 @@
+            staleSegment);
+        Map<String, SegmentFileStore.FolderDetails> locationMap = fileStore.getSegmentFile()
+            .getLocationMap();
+        if (locationMap != null) {
+          CarbonFile segmentLocation = FileFactory.getCarbonFile(carbonTable.getTablePath() +
+              CarbonCommonConstants.FILE_SEPARATOR + fileStore.getSegmentFile().getLocationMap()
+              .entrySet().iterator().next().getKey());
+          // copy the complete segment to the trash folder
+          TrashUtil.copySegmentToTrash(segmentLocation, CarbonTablePath.getTrashFolderPath(
+              carbonTable.getTablePath()) + CarbonCommonConstants.FILE_SEPARATOR +
+              timeStampForTrashFolder + CarbonCommonConstants.FILE_SEPARATOR + CarbonTablePath
+              .SEGMENT_PREFIX + segmentNumber);
+          // Deleting the stale Segment folders.
+          try {
+            CarbonUtil.deleteFoldersAndFiles(segmentLocation);
+          } catch (IOException | InterruptedException e) {
+            LOGGER.error("Unable to delete the segment: " + segmentNumber + " from after moving" +
+                " it to the trash folder : " + e.getMessage(), e);
+          }
+          // delete the segment file as well
+          FileFactory.deleteFile(CarbonTablePath.getSegmentFilePath(carbonTable.getTablePath(),
+              staleSegment));
+        }
+      }
+      staleSegments.clear();
+    }
+  }
+
+  /**
+   * This method will clean all the stale segments for partition table, delete the source folders
+   * after copying the data to the trash and also remove the .segment files of the stale segments
+   */
+  public static void cleanStaleSegmentsForPartitionTable(CarbonTable carbonTable)
+    throws IOException {
+    long timeStampForTrashFolder = System.currentTimeMillis();
+    List<String> staleSegments = getStaleSegments(carbonTable);
+    if (staleSegments.size() > 0) {
+      for (String staleSegment : staleSegments) {
+        String segmentNumber = staleSegment.split(CarbonCommonConstants.UNDERSCORE)[0];
+        // for each segment we get the indexfile first, then we get the carbondata file. Move both
+        // of those to trash folder
+        List<CarbonFile> filesToDelete = new ArrayList<>();
+        SegmentFileStore fileStore = new SegmentFileStore(carbonTable.getTablePath(),
+            staleSegment);
+        List<String> indexOrMergeFiles = fileStore.readIndexFiles(SegmentStatus.SUCCESS, true,
+            FileFactory.getConfiguration());
+        for (String file : indexOrMergeFiles) {
+          // copy the index or merge file to the trash folder
+          TrashUtil.copyFileToTrashFolder(file, CarbonTablePath.getTrashFolderPath(carbonTable
+              .getTablePath()) + CarbonCommonConstants.FILE_SEPARATOR + timeStampForTrashFolder +
+              CarbonCommonConstants.FILE_SEPARATOR + CarbonTablePath.SEGMENT_PREFIX +
+              segmentNumber);
+          filesToDelete.add(FileFactory.getCarbonFile(file));
```

Comment:
need to fail the clean files command, if copy fails. so directly throwing exception if anything fails. In case of copy failure, it will not delete anything from the source data files